### DISCUSSION TITLE
Attach a MessageQueue.hnd to view_port_input_callback_set

### DISCUSSION THREAD

**Discussion by kpmy** [dcoles/flipperzero-rs#87] [Q&A]
Hi!

What is the best way to proper use `MessageQueue`? For now, I had to modify `hnd` property to pub, then doing something like this. It works, but seems dirty for me.
```rust

pub unsafe extern "C" fn input_callback(event: *mut InputEvent, context: *mut c_void){
    furi_message_queue_put(context, event as *const c_void, duration_to_ticks(Duration::from_secs(1))); //TOO LOW LEVEL
}

fn main(_args: *mut u8) -> i32 {
    let event_queue: MessageQueue<InputEvent> = MessageQueue::new(8);
    let view_port: *mut ViewPort;
    unsafe {
        view_port = view_port_alloc();
        //...
        view_port_input_callback_set(view_port, Some(input_callback), event_queue.hnd.as_ptr()); //THIS
       //...
```

Best thing I figured is `static event_queue: Option...` and then initialize, but dirty code with `.hnd` still stays.

**Comment by JarvisCraft** [comment]
`ViewPort`-related safe bindings are currently under development at #29 so as for now, unfortunately, the right way is using direct (unsafe) bindings.
